packages/ajax/test/AjaxClass.interceptors-cache.test.js

```diff
@@ -0,0 +1,713 @@
+      .then(() => {
+        expect(server.requests.length).to.equal(4);
+      })
+      .then(() => ajax.get('/posts'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .then(() => ajax.get('/posts/1'))
+      .then(() => {
+        expect(server.requests.length).to.equal(5);
+      })
+      .finally(() => {
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('does not returns the cached object on second call default timeToLive`', () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 0,
+    });
+
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        const clock = sinon.useFakeTimers();
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+        clock.tick(1);
+        clock.restore();
+      })
+      .then(() => ajax.get('/test'))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it("returns the cached object on second call with `useCache: 'always'`, with querystring parameters", () => {
+    newCacheId();
+
+    const indexes = addCacheInterceptors(ajax, {
+      useCache: 'always',
+      timeToLive: 100,
+    });
+
+    const ajaxGetSpy = sinon.spy(ajax, 'get');
+
+    return ajax
+      .get('/test', {
+        params: {
+          q: 'test',
+          page: 1,
+        },
+      })
+      .then(() => {
+        expect(ajaxGetSpy.calledOnce).to.be.true;
+        expect(ajaxGetSpy.calledWith('/test')).to.be.true;
+      })
+      .then(() =>
+        ajax.get('/test', {
+          params: {
+            q: 'test',
+            page: 1,
+          },
+        }),
+      )
+      .then(() => {
+        expect(server.requests.length).to.equal(1);
+      })
+      .finally(() => {
+        ajaxGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it("makes two requests when using `useCache: 'never'` in the action", () => {
+    newCacheId();
+    getCacheIdentifier = () => 'cacheIdentifier2';
+
+    const ajaxAlwaysGetSpy = sinon.spy(ajax, 'get');
+    const indexes = addCacheInterceptors(ajax, { useCache: 'always' });
+
+    return ajax
+      .get('/test')
+      .then(() => {
+        expect(ajaxAlwaysGetSpy.calledOnce, 'calledOnce').to.be.true;
+        expect(ajaxAlwaysGetSpy.calledWith('/test'));
+      })
+      .then(() => ajax.get('/test', { lionCacheOptions: { useCache: 'never' } }))
+      .then(() => {
+        expect(server.requests.length).to.equal(2);
+      })
+      .finally(() => {
+        ajaxAlwaysGetSpy.restore();
+        removeCacheInterceptors(ajax, indexes);
+      });
+  });
+
+  it('allows direct ajax calls without cache interceptors configured', () => {
```

Comment:
Maybe put this test at the top, since it's kinda a smoke test of the default ajax behavior without caching.